src/osdc/Objecter.cc

```diff
@@ -1944,7 +1936,7 @@ void Objecter::_maybe_request_map()
     flag = CEPH_SUBSCRIBE_ONETIME;
   }
   epoch_t epoch = osdmap->get_epoch() ? osdmap->get_epoch()+1 : 0;
-  if (monc->sub_want("osdmap", epoch, flag)) {
+  if (monc->sub_want("osdmap", epoch, flag | CEPH_SUBSCRIBE_LATEST)) {
```

Comment:
So the new flag is still set unconditionally in Objecter?  As soon as someone changes mon_max_span_of_incremental from its default of 0, all Objecter users would be affected.  This doesn't match the description you gave earlier

> The solution of this pr is to add a flag(CEPH_SUBSCRIBE_LATEST) to indicate whether the client can handle a full map and skip the intermediate versions.
> If the client's request doesn’t carry the flag, mon will follow the original processing method
> If the client has this flag, it means that the client can accept a full osdmap, and mon will measure the cost of the full osdmap or the incremental osdmap according to the version gap of the client.
> The modification on mon is mainly to recognize this flag.

because all requests would carry the flag.  The commit is titled "client : make all client always request for latest version osdmap", but once again, Objecter is used by servers too.

Reply:
@idryomov  I added a parameter ( objecter_want_latest ) to allow the objecter to choose whether to use the flag, CEPH_SUBSCRIBE_LATEST, or not. Thx.